### PR TITLE
Add support for build args in composefiles

### LINKED ISSUE #214: Support build args in composefiles
You can now pass arguments to a build if you're using the new file format:

```yaml
build:
  context: .
  args:
    buildno: 1
```

### PR DESCRIPTION
**Pull request by vdemeester**
This connect the dots to support build args from composefiles 🐳.

     build:
       context: .
       args:
         buildno: 1

The composefile part above will behave the same as `docker build --build-arg buildno=1`.

/cc @dnephin @joshwget @ibuildthecloud 

Closes #214 

🐸
Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Comment by joshwget** [comment]
LGTM. `ToMap` will likely be handy in the future as well.

**Comment by vdemeester** [comment]
I don't understand the failure on jenkins.. hum..

File: script/.integration-daemon-start [hunk 1]
```diff
@@ -11,7 +11,7 @@ fi
 # intentionally open a couple bogus file descriptors to help test that they get scrubbed in containers
 exec 41>&1 42>&2
 
-export DOCKER_GRAPH=${DOCKER_GRAPH:-/var/lib/docker/${DOCKER_DAEMON_VERSION}}
```
Comment (vdemeester) [inline script/.integration-daemon-start]: This wasn't working, the second time, `DOCKER_GRAPH` was exported :sweat_smile:

**Comment by vdemeester** [comment]
Finally it's green :green_heart:

**Pull request merged**